app/views/ReportView/components/Settings/components/ReportHistory/index.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useContext, useEffect, useState } from 'react';
+import {
+  Typography,
+} from '@mui/material';
+
+import api from '@/services/api';
+import ReportContext from '@/context/ReportContext';
+import useResource from '@/hooks/useResource';
+import { formatDate } from '@/utils/date';
+
+import './index.scss';
+import { useQuery } from 'react-query';
+
+const ReportHistory = (): JSX.Element => {
+  const { report } = useContext(ReportContext);
+  let { reportEditAccess: canEdit } = useResource();
+  const [orderedHistory, setOrderedHistory] = useState([]);
+  if (report.state === 'completed') {
+    canEdit = false;
+  }
+
+  const { data: signatureHistoryData } = useQuery(
+    `/reports/${report.ident}/signatures/history`,
+    async ({queryKey: [route] }) => await api.get(route).request(),
+    {
+      staleTime: Infinity,
+      select: (response) => {
+        return response
+      },
+    },
+  );
+
+  const { data: stateHistoryData } = useQuery(
+    `/reports/${report.ident}/state-history`,
+    async ({queryKey: [route] }) => await api.get(route).request(),
+    {
+      staleTime: Infinity,
+      select: (response) => {
+        return response;
+      },
+    },
+  );
+
+  useEffect(() => {
+    if (signatureHistoryData && stateHistoryData && report.users) {
+      const historyArr = signatureHistoryData.concat(stateHistoryData).concat(report.users);
+      historyArr.sort(function(a,b){
+        return new Date(a.updatedAt).valueOf() - new Date(b.updatedAt).valueOf();
+      });
+      const orderedHistoryArr = handleOrderReportHistory(historyArr);
+      setOrderedHistory(orderedHistoryArr);
+    }
+  }, [signatureHistoryData, stateHistoryData, report.users]);
+
+  const handleOrderReportHistory = (orderedHistory) => {
+    const historyArray = [];
+    let authorSigned = null;
+    let creatorSigned = null;
+    let reviewerSigned = null;
+
+    for (const element of orderedHistory) {
+      if (element.hasOwnProperty('authorSignature')) {
```

Comment:
More linting really
`Object.prototype.hasOwnProperty.call(element, 'authorSignature')` is a workaround for the [no-prototype-builtins](https://eslint.org/docs/latest/rules/no-prototype-builtins) rule, but we COULD update the compiler option in `tsconfig.json` to `"lib": ["ES2022", "DOM"]` so then you can use `Object.hasOwn(element, 'authorSignature')` instead